code-challenges.test.js

```diff
@@ -22,13 +22,34 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+describe("When wordDecoder is given a string,", () => {
 
+    it("returns a coded message with numbers in place of vowels: 4 for a, 3 for e, 1 for i, and 0 for o.", () => {
+        
+        const actResult = wordDecoder(secretCodeWord1)
 
+        expect(actResult).toEqual("l4ck4d41s1c4l")
 
+    })
+    it("returns a coded message with numbers in place of vowels: 4 for a, 3 for e, 1 for i, and 0 for o.", () => {
+
+        const actResult = wordDecoder(secretCodeWord2)
+
+        expect(actResult).toEqual("g0bbl3dyg00k")
+
+    })
+})
 
 // b) Create the function that makes the test pass.
+const wordDecoder = (string) => {
 
+    return string.replaceAll('a', 4).replaceAll('e', 3).replaceAll('i', 1).replaceAll('o', 0)
```

Comment:
higher order functions for the win! And also, it looks like `.replaceAll()` is in a future version of JS. Great exploration of new features!